src/simulators/statevector/qubitvector_thrust.hpp

```diff
@@ -4797,22 +4800,22 @@ double QubitVectorThrust<data_t>::expval_pauli(const reg_t &qubits,
 
   // Compute the overall phase of the operator.
   // This is (-1j) ** number of Y terms modulo 4
-  thrust::complex<data_t> phase(1,0);
+  auto phase = thrust::complex<data_t>(coeff);
   switch (num_y & 3) {
     case 0:
       // phase = 1
       break;
     case 1:
       // phase = -1j
-      phase = thrust::complex<data_t>(0, -1);
+      phase = thrust::complex<data_t>(phase.imag(), -phase.real());
       break;
     case 2:
       // phase = -1
-      phase = thrust::complex<data_t>(-1, 0);
+      phase = thrust::complex<data_t>(-phase.real(), -phase.imag());
       break;
     case 3:
       // phase = 1j
-      phase = thrust::complex<data_t>(0, 1);
+      phase = thrust::complex<data_t>(-phase.imag(), phase.real());
       break;
   }
```

Comment:
If I'm not wrong we can use `add_y_phase` function defined in `qubitvector` adding a template parameter:

```
template <template<class> typename complex_type>
void add_y_phase(uint_t num_y, complex_type<data_t>& coeff){ .... }
```

Reply:
Only thing is QVThrust cant access functions of QV (its basically a duplicate class), so it would just be adding that function to this class as well.

Reply:
right, I wrongly assumed it was a derived class of QubitVector. I guess there are more chunks of code duplicated so I think we can add the function here as well, and at some point try to put the common code in a place accesible to both classes.